config/src/main/java/org/springframework/security/config/web/server/ServerHttpSecurity.java

```diff
@@ -993,6 +995,19 @@ public OAuth2LoginSpec authenticationManager(ReactiveAuthenticationManager authe
 			return this;
 		}
 
+		/**
+		 * The {@link ServerSecurityContextRepository} used to save the {@code Authentication}. Defaults to
+		 * {@link WebSessionServerSecurityContextRepository}.
+		 *
+		 * @since 5.2
+		 * @param securityContextRepository the repository to use
+		 * @return the {@link OAuth2LoginSpec} to continue configuring
+		 */
+		public OAuth2LoginSpec securityContextRepository(ServerSecurityContextRepository securityContextRepository) {
```

Comment:
Would you please add a simple test to `OAuth2LoginTests` that confirms that an application-provided security context repository is correctly picked up?

Reply:
I modified an existing test, I hope it's fine enough.